sympy/sets/handlers/intersection.py

```diff
@@ -306,17 +306,17 @@ def intersection_sets(self, other): # noqa:F811
             # univarite imaginary part in same variable
             x, xis = zip(*[solve_linear(i, 0) for i in Mul.make_args(im) if n in i.free_symbols])
             if x and all(i == n for i in x):
-                base_set = FiniteSet(*xis)
+                base_set &= FiniteSet(*xis)
             else:
-                base_set -= ConditionSet(n, Eq(im, 0), S.Integers)
+                base_set = ConditionSet(n, Eq(im, 0), base_set)
         # exclude values that make denominators 0
         for i in denoms(f):
             if i.has(n):
                 sol = list(zip(*[solve_linear(i, 0) for i in Mul.make_args(im) if n in i.free_symbols]))
                 if sol != []:
                     x, xis = sol
                     if x and all(i == n for i in x):
-                        base_set -= FiniteSet(xis)
+                        base_set &= FiniteSet(*xis)
```

Comment:
This one was correct -- we must exclude from the base set any solutions of denoms. Check what you get for `Lambda(n, n/3 + (n - 2)*(n - 1)*I/(2*n - 2), Integers).intersection(Reals)` -- the answer should be `{2/3}`

Reply:
ImageSet(Lambda(x, x/2), ConditionSet(n, Eq((n - 2)*(n - 1)/(2*n - 2), 0), Integers)) I get this with the &=

Reply:
```
In [1]: s1=imageset(Lambda(n, n/2 + (n - 2)*(n - 1)*I/(2*n - 2)), Integers)

In [2]: s1.intersect(Reals)
Out[2]: 
⎧x │     ⎧  │         ⎛(n - 2)⋅(n - 1)    ⎞⎫⎫
⎨─ │ x ∊ ⎨n │ n ∊ ℤ ∧ ⎜─────────────── = 0⎟⎬⎬
⎩2 │     ⎩  │         ⎝    2⋅n - 2        ⎠⎭⎭

```